### PR TITLE
[21459] Add IDL view

### PR DESCRIPTION
**Pull request by Carlosespicur**
This PR adds a new view to show the IDL representation of a type discovered in logical view and graph.

**Needed to be merged after:**
- https://github.com/eProsima/Fast-DDS-statistics-backend/pull/239
- https://github.com/eProsima/Fast-DDS-monitor/pull/227

**Related PR**
- https://github.com/eProsima/Fast-DDS-statistics-backend/pull/245

### CODE REVIEW

File: src/Controller.cpp [hunk 1]
```diff
@@ -322,3 +322,15 @@ QString Controller::get_domain_view_graph(
...
+        QString entity_id)
+{
+    return utils::to_QString(engine_->get_type_idl(backend::models_id_to_backend_id(entity_id)));
+}
+
+QString Controller::get_data_type_name(
+        QString entity_id)
+{
+    return utils::to_QString(engine_->get_data_type_name(backend::models_id_to_backend_id(entity_id)));
+}
```
Comment (JesusPoderoso) [inline src/Controller.cpp:336]: EOF

File: src/backend/SyncBackendConnection.cpp [hunk 2]
```diff
@@ -953,6 +959,21 @@ Graph SyncBackendConnection::get_domain_view_graph (
...
+    {
+        return StatisticsBackend::get_type_idl(id);
+    }
+    catch(const std::exception& e)
+    {
+        qWarning() << "Fail getting the IDL type for entity id " << id.value() << ": " << e.what();
+        return "";
+    }
+    
+}
```
Comment (JesusPoderoso) [inline src/backend/SyncBackendConnection.cpp:975]: Remove line break
```suggestion
    }
}
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
I've generated custom types using Fast DDS Xtypes example to check the idl representation bounds in the tab.
1) Vertically, the scrollbar is so thin when displayed, and mousewheel scrolling is disabled. This should be fixed.
2) Horizontally, the text is not elided when out of bounds.

**Review by Carlosespicur — COMMENTED** [review]
Commits reviewed

File: qml/TabLayout.qml [hunk 3]
```diff
@@ -295,9 +299,56 @@ Item {
                         boundsBehavior: Flickable.StopAtBounds
                         interactive: false
                         anchors.fill: parent
+                        width: parent.width
+                        height: parent.height
                         contentWidth: parent.width
                         contentHeight: idl_text.height + 2 * tabLayout.idl_text_margin_
-                        ScrollBar.vertical: ScrollBar {}
+                        ScrollBar.vertical: ScrollBar {
```
Comment (JesusPoderoso) [inline qml/TabLayout.qml:306]: Fix identation in all this included code

File: qml/TabLayout.qml [hunk 4]
```diff
@@ -279,6 +290,98 @@ Item {
...
+                        width: parent.width
+                        height: parent.height
+                        contentWidth: parent.width
+                        contentHeight: idl_text.height + 2 * tabLayout.idl_text_margin_
+                        ScrollBar.vertical: ScrollBar {
+                        id: vertical_bar
+                        policy: ScrollBar.AlwaysOn
+                        stepSize: tabLayout.scroll_speed_
+                        anchors.top: parent.top;        anchors.topMargin: 0
+                        anchors.right: parent.right;    anchors.rightMargin: tabLayout.idl_text_margin_/10
```
Comment (JesusPoderoso) [inline qml/TabLayout.qml:311]: Remove right margin
```suggestion
                        anchors.right: parent.right
```

File: qml/TabLayout.qml [hunk 5]
```diff
@@ -279,6 +290,98 @@ Item {
...
+                        boundsBehavior: Flickable.StopAtBounds
+                        interactive: false
+                        anchors.fill: parent
+                        width: parent.width
+                        height: parent.height
+                        contentWidth: parent.width
+                        contentHeight: idl_text.height + 2 * tabLayout.idl_text_margin_
+                        ScrollBar.vertical: ScrollBar {
+                        id: vertical_bar
+                        policy: ScrollBar.AlwaysOn
```
Comment (JesusPoderoso) [inline qml/TabLayout.qml:308]: I would make it visible only if necessary:
```suggestion
                        policy: ScrollBar.AlwaysOn
                        visible: idlView.contentHeight > idlView.height
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
A couple of details

**Review by JesusPoderoso — APPROVED** [review]
LGTM with green CI

**Pull request merged**